### PR TITLE
deploy: Add skipForceFlatten for daemonset nodeplugin

### PR DESCRIPTION
**Pull request by muxuelanKK**
# Describe what this PR does #

Provide some context for the reviewer

## Is there anything that requires special attention ##

Do you have any questions?

Is the change backward compatible?

Are there concerns around backward compatibility?

Provide any external context for the change, if any.

For example:

* Kubernetes links that explain why the change is required
* CSI spec related changes/catch-up that necessitates this patch
* golang related practices that necessitates this change

## Related issues ##

Mention any github issues relevant to this PR. Adding below line
will help to auto close the issue once the PR is merged.

Fixes: #issue_number

## Future concerns ##

List items that are not part of the PR and do not impact it's
functionality, but are work items that can be taken up subsequently.

**Checklist:**

* [ ] **Commit Message Formatting**: Commit titles and messages follow
  guidelines in the [developer
  guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [ ] Reviewed the developer guide on [Submitting a Pull
  Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [ ] [Pending release
  notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md)
  updated with breaking and/or notable changes for the next major release.
* [ ] Documentation has been updated, if necessary.
* [ ] Unit tests have been added, if necessary.
* [ ] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

### CODE REVIEW

**Comment by nixpanic** [comment]
Hi @muxuelanKK, thanks for the PR!

Could you update the title of your commit to something like

    deploy: Add skipForceFlatten for daemonset nodeplugin

A keyword like `deploy`, or `rbd` is required so that it is easier for reviewers to understand what the PR/commit is about. Once changing in your local branch, make sure to force-push the single commit to your GitHub repo/branch.

Currently the PR description does not contain anything useful. You can either use the template that is present now, or describe the need for the change in free form (it is good practice to include a description in the commit too).

File: charts/ceph-csi-rbd/values.yaml [hunk 1]
```diff
@@ -98,6 +98,19 @@ nodeplugin:
...
+  # Hard limit for maximum number of nested volume clones that are taken before
+  # a flatten occurs
+  hardMaxCloneDepth: 8
+  # Soft limit for maximum number of nested volume clones that are taken before
+  # a flatten occurs
+  softMaxCloneDepth: 4
+  # Maximum number of snapshots allowed on rbd image without flattening
+  maxSnapshotsOnImage: 450
+  # Minimum number of snapshots allowed on rbd image to trigger flattening
+  minSnapshotsOnImage: 250
```
Comment (Madhu-1) [inline charts/ceph-csi-rbd/values.yaml:110]: can we comment these out instead of having default here as well?

Comment (Madhu-1) [inline charts/ceph-csi-rbd/values.yaml:110]: These values can be set for both controller and nodeplugin, lets have it in a common place and use it in both controller and plugins

File: charts/ceph-csi-rbd/templates/nodeplugin-daemonset.yaml [hunk 2]
```diff
@@ -57,6 +57,13 @@ spec:
             - "--csi-addons-endpoint=$(CSI_ADDONS_ENDPOINT)"
             - "--v={{ .Values.logLevel }}"
             - "--drivername=$(DRIVER_NAME)"
+            - "--rbdhardmaxclonedepth={{ .Values.nodeplugin.hardMaxCloneDepth }}"
+            - "--rbdsoftmaxclonedepth={{ .Values.nodeplugin.softMaxCloneDepth }}"
+            - "--maxsnapshotsonimage={{ .Values.nodeplugin.maxSnapshotsOnImage }}"
+            - "--minsnapshotsonimage={{ .Values.nodeplugin.minSnapshotsOnImage }}"
```
Comment (Madhu-1) [inline charts/ceph-csi-rbd/templates/nodeplugin-daemonset.yaml:63]: can you please make similar changes to deployment as well?

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
Please update the documentation as well for the new fields

File: charts/ceph-csi-rbd/templates/nodeplugin-daemonset.yaml (hunk 2 above)
Comment (muxuelanKK) [reply charts/ceph-csi-rbd/templates/nodeplugin-daemonset.yaml:63]: Deployment already had, it's no need to add.

**Pull request closed without merging**